compiler/typecheck/TcDaml.hs

```diff
@@ -193,24 +203,29 @@ displayError info TriedImplementNonInterface { triedIface }
            text ", but" <+> pprq triedIface <+> text "is not an interface."
 
 dedupe :: DamlInfo -> DamlInfo
-dedupe (DamlInfo x0 x1 x2 x3 x4 x5) =
+dedupe (DamlInfo x0 x1 x2 x3 x4 x5 x6) =
   DamlInfo
-    (nubSort x0)
-    (nubSort x1)
-    (nubSortBy (\(ifaceOrTpl, choice, type_) -> (ifaceOrTpl, choice)) x2)
+    x0
+    x1
+    x2
     (nubSortBy (\(mName, (cName, type_)) -> (mName, cName)) x3)
     (nubSort x4)
-    (nubSortBy fst x5)
+    x5
+    x6
   where
     nubSortBy :: Ord b => (a -> b) -> [a] -> [a]
     nubSortBy f xs = M.elems $ M.fromList $ map (\x -> (f x, x)) xs
 
 pprq :: Outputable a => a -> SDoc
 pprq = quotes . ppr
 
-isTemplate, isInterface :: DamlInfo -> Name -> Bool
-isTemplate info name = name `elem` templates info
-isInterface info name = name `elem` interfaces info
+synEq :: DamlInfo -> Name -> Name -> Bool
+synEq info n1 n2 = resolveSynonym info n1 == resolveSynonym info n2
+
+isTemplate, isInterface, isSynonym :: DamlInfo -> Name -> Bool
+isTemplate info name = resolveSynonym info name `S.member` templates info
+isInterface info name = resolveSynonym info name `S.member` interfaces info
+isSynonym info name = resolveSynonym info name `M.member` synonyms info
```

Comment:
not important since this isn't used, but I think we _shouldn't_ resolve synonyms on this one but rather just check membership

Reply:
I was hoping to have synonyms be completely transparent everywhere, except for *maybe* in pretty printing, so that I don't have to worry about a check missing interfaces and templates due to synonyms.

I can see that the `isSynonym` check is broken right now, I can fix that - is there something else I'm missing?